src/lib/operators/projection.cpp

```diff
@@ -97,47 +97,109 @@ std::shared_ptr<const Table> Projection::_on_execute() {
     step_performance_data.set_step_runtime(OperatorSteps::UncorrelatedSubqueries, timer.lap());
   }
 
+  auto forwarding_cost = std::chrono::nanoseconds{};
+  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+
+  const auto chunk_count = input_table.chunk_count();
+
   // Perform the actual projection on a per-chunk level. `output_segments_by_chunk` will contain both forwarded and
   // newly generated columns. In the upcoming loop, we do not yet deal with the projection_result_table indirection
   // described above.
-  auto output_segments_by_chunk = std::vector<Segments>(input_table.chunk_count());
+  auto output_segments_by_chunk = std::vector<Segments>(chunk_count);
 
-  auto forwarding_cost = std::chrono::nanoseconds{};
-  auto expression_evaluator_cost = std::chrono::nanoseconds{};
+  // Create a vector that will hold all jobs that are getting executed in parallel.
+  auto jobs = std::vector<std::shared_ptr<AbstractTask>>{};
+  jobs.reserve(chunk_count);
 
-  const auto chunk_count = input_table.chunk_count();
+  const auto expression_count = expressions.size();
+
+  // Create a two-dimensional vector that saves the information if a column is nullable for every segment.
+  // This allows parallel write operation per thread. Later this data structure is aggregated
+  // to a vector with information if a column is nullable for every column.
+  auto column_is_nullable_by_chunk = std::vector<std::vector<bool>>(
```

Comment:
If your first dimension is the chunk id, wouldn't it be `segment_is_nullable`? In any case, are you just adding the chunk dimension so that threads can update this data in parallel? Maybe a `vector<atomic_bool>` would do the same job with less code?

Reply:
I agree `segment_is_nullable` would be a better choice. I will have a look at `vector<atomic_bool>`.